src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```
@@ -5,10 +5,31 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_LENGTH_PASSWORD = 6;
+    public static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("Parameter User has a null value");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Field Age has a null value");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Field Password has a null value");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Field Login has a null value");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("Age less than " + MIN_AGE);
+        }
+        if (user.getPassword().length() < MIN_LENGTH_PASSWORD) {
+            throw new RegistrationException("Password length less than " + MIN_LENGTH_PASSWORD);
+        }
+        // if retrying user registration (login already exists) - return null
+        return storageDao.get(user.getLogin()) == null ? storageDao.add(user) : null;
```

let's throw an exception if user already exists